networkx/algorithms/tests/test_hierarchy_coordinates.py

```diff
@@ -0,0 +1,225 @@
+    last_hash = False
+    for G in Gs:
+        if last_hash:
+            assert last_hash == nx.weisfeiler_lehman_graph_hash(G)
+        last_hash = nx.weisfeiler_lehman_graph_hash(G)
+
+
+def test_max_min_layers():
+    G = nx.DiGraph()
+    G.add_edges_from([(0, 1), (0, 2), (1, 3), (1, 4), (2, 4), (2, 5)])
+    assert nx.max_min_layers(G, max_layer=True) == [0]
+    assert nx.max_min_layers(G, max_layer=False) == [3, 4, 5]
+
+    # G = nx.cycle_graph(5, create_using=nx.DiGraph)  # cannot detect cycles with just DAG check...
+    # pytest.raises(nx.NetworkXError, nx.max_min_layers, G)
+
+
+def test_leaf_removal():
+    G = nx.DiGraph()
+    G.add_edges_from([(0, 1), (0, 2), (1, 3), (1, 4), (2, 4), (2, 5)])
+
+    bottomless_G = nx.DiGraph()
+    bottomless_G.add_edges_from([(0, 1), (0, 2)])
+    assert nx.leaf_removal(G, top=False).edges() == bottomless_G.edges()
+    assert nx.leaf_removal(G, top=False).nodes() == bottomless_G.nodes()
+
+    topless_G = nx.DiGraph()
+    topless_G.add_edges_from([(1, 3), (1, 4), (2, 4), (2, 5)])
+    assert nx.leaf_removal(G, top=True).edges() == topless_G.edges()
+    assert nx.leaf_removal(G, top=True).nodes() == topless_G.nodes()
+
+    G = nx.cycle_graph(5, create_using=nx.DiGraph)
+    pytest.raises(nx.NetworkXError, nx.leaf_removal, G)
+
+
+def test_recursive_leaf_removal():
+    diamond_net = [
+        (0, 1),
+        (0, 2),
+        (1, 3),
+        (1, 4),
+        (2, 4),
+        (2, 5),
+        (3, 6),
+        (4, 6),
+        (4, 7),
+        (5, 7),
+        (6, 8),
+        (7, 8),
+    ]
+    layer_0 = [(0, 1), (0, 2)]
+    layer_1 = [(1, 3), (2, 4), (1, 4), (2, 5)]
+    layer_2 = [(3, 6), (4, 6), (4, 7), (5, 7)]
+    layer_3 = [(6, 8), (7, 8)]
+    G = nx.DiGraph(diamond_net)
+
+    pruned_from_top = nx.recursive_leaf_removal(G, from_top=True)
+    pruned_from_bottom = nx.recursive_leaf_removal(G, from_top=False)
+
+    compare_hashed_graphs(
+        pruned_from_top[0], pruned_from_bottom[0]
+    )  # As the original graph is always the first element
+    compare_hashed_graphs(pruned_from_top[1], nx.DiGraph(layer_1 + layer_2 + layer_3))
+    compare_hashed_graphs(pruned_from_top[2], nx.DiGraph(layer_2 + layer_3))
+    compare_hashed_graphs(pruned_from_top[3], nx.DiGraph(layer_3))
+    compare_hashed_graphs(
+        pruned_from_bottom[1], nx.DiGraph(layer_2 + layer_1 + layer_0)
+    )
+    compare_hashed_graphs(pruned_from_bottom[2], nx.DiGraph(layer_1 + layer_0))
+    compare_hashed_graphs(pruned_from_bottom[3], nx.DiGraph(layer_0))
+
+    G = nx.cycle_graph(5, create_using=nx.DiGraph)  # No cycles
+    pytest.raises(nx.NetworkXError, nx.recursive_leaf_removal, G)
+
+
+def test_weight_nodes_by_condensation():
+    center_cycle = [(0, 1), (1, 2), (2, 3), (3, 1), (3, 4)]
+    G = nx.DiGraph(center_cycle)  # unit weights by default
+    pytest.raises(
+        nx.NetworkXError, nx.weight_nodes_by_condensation, G
+    )  # Must be condensed (acyclic)
```

Comment:
Can you move this comment to the line prior to the pytest?
And other comments in this file that end a line...   Thanks!